### PR TITLE
PWA: Only initialize resources service when element is attached to DOM

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

File: src/base-element.js [hunk 1]
```diff
@@ -172,6 +172,15 @@ export class BaseElement {
     return this.win;
   }
 
+  /**
+   * Returns the associated ampdoc. Only available when `buildCallback` and
```
Comment (aghassemi) [inline src/base-element.js]: nit: mention it throws if accessed before buildCallback (I had to look at implementation to see if I get null or exception, so worth documenting)

File: src/custom-element.js [hunk 2]
```diff
@@ -490,13 +500,25 @@ function createBaseAmpElementProto(win) {
...
 
   /**
+   * Returns the associated ampdoc. Only available after attachment.
+   * @return {!./service/ampdoc-impl.AmpDoc}
+   * @final @this {!Element}
+   * @package
+   */
+  ElementProto.getAmpDoc = function() {
+    return /** @type {!./service/ampdoc-impl.AmpDoc} */ (
+        dev().assert(this.ampdoc_, 'no ampdoc yet'));
```
Comment (aghassemi) [inline src/custom-element.js]: nit: `no amodoc yet, since element is not attached`

File: src/base-element.js (hunk 1 above)
Comment (dvoytenko) [inline src/base-element.js]: Done

File: src/custom-element.js (hunk 2 above)
Comment (dvoytenko) [inline src/custom-element.js]: Done

**Pull request merged**